node-upstream-tests/run_tests.js

```diff
@@ -0,0 +1,387 @@
+'use strict'
+
+/* eslint-disable no-console */
+
+const promisify = require('util').promisify
+const axios = require('axios')
+const childProcess = require('child_process')
+const fs = require('fs')
+const glob = promisify(require('glob'))
+const path = require('path')
+
+const NODE_BIN = process.env['NODE_BIN'] || '/usr/bin/node'
+const NODE_REPO_PATH = process.env['NODE_PROJECT']
+if (NODE_REPO_PATH === undefined) {
+  throw new Error('The env variable NODE_PROJECT is not set. This is required to locate the root of the nodejs repo')
+}
+
+const NEEDS_TO_SPAWN_AGENT = false
```

Comment:
As mentioned in another comment, let's leave the test agent out of this PR and revisit later so that this PR can focus only on making sure Node tests are passing.